ble-aranet4.js

```diff
@@ -0,0 +1,229 @@
+      jmp = 0;
+      bufFn = null;
+      if (fmt[pos] === 'b' || fmt[pos] === 'B') jmp = 1;
+      if (fmt[pos] === 'h' || fmt[pos] === 'H') jmp = 2;
+      if (fmt[pos] === 'b') {
+        res[keyArr[pos]] = this.getInt8();
+      }
+      else if (fmt[pos] === 'B') {
+        res[keyArr[pos]] = this.getUInt8();
+      }
+      else if (fmt[pos] === 'h') {
+        res[keyArr[pos]] = le ? this.getInt16LE() : this.getInt16BE();
+      }
+      else if (fmt[pos] === 'H') {
+        res[keyArr[pos]] = le ? this.getUInt16LE() : this.getUInt16BE();
+      }
+      this.buffer = this.buffer.slice(jmp);
+      pos++;
+    }
+    return res;
+  }
+};
+
+let Aranet4Parser = {
+  lastPC: null,
+  getData: function (res) {
+    let rm = null;
+    let mfd_data = BLE.GAP.ParseManufacturerData(res.advData);
+    if (typeof mfd_data !== "string" || mfd_data.length < 8) return null;
+    packedStruct.setBuffer(mfd_data);
+    let hdr = packedStruct.unpack('<HB', ['mfd_id', 'status']);
+    if (hdr.mfd_id !== ARANET_MFD_ID) return null;
+
+    let aranet_status = {
+      integration: (hdr.status & (1 << 5)) !== 0,
+      dfu: (hdr.status & (1 << 4)) !== 0,
+      cal_state: (hdr.status >> 1) & 3
+    };
+
+    let aranet_sys_data = packedStruct.unpack('<BBHB', [
+      'fw_patch',
+      'fw_minor',
+      'fw_major',
+      'hw'
+    ]);
+
+    aranet_sys_data.addr = res.addr;
+    aranet_sys_data.rssi = res.rssi;
+
+    rm = {
+      status: aranet_status,
+      sys: aranet_sys_data
+    };
+
+    if (!aranet_status.integration) return rm;
+
+    let aranet_data = packedStruct.unpack('<BBHHHBBBHHB', [
+      'region',
+      'packaging',
+      'co2_ppm',
+      'tC',
+      'pressure_dPa',
+      'rh',
+      'battery',
+      'co2_aranet_level',
+      'refresh_interval',
+      'age',
+      'packet_counter'
+    ]);
+
+    //skip if PC has not changed
+    if (CONFIG.dedupe && this.lastPC !== null && this.lastPC === aranet_data.packet_counter) return null;
+
+    this.lastPC = aranet_data.packet_counter;
+
+    aranet_data.tC = aranet_data.tC / 20.0;
+
+    let dIdx;
+    for(dIdx in aranet_data) {
+      rm[dIdx] = aranet_data[dIdx];
+    }
+
+    return rm;
+  }
+};
+
+function scanCB(ev, res) {
+  if (ev !== BLE.Scanner.SCAN_RESULT) return;
+  let measurement = Aranet4Parser.getData(res);
+  if (measurement === null) return;
+  print("aranet measurement:", JSON.stringify(measurement));
+
+  // execute actions from CONFIG
+  let aIdx = null;
+  for (aIdx in CONFIG.actions) {
+    // skip if no condition defined
+    if (typeof CONFIG.actions[aIdx]["cond"] === "undefined") continue;
+    let cond = CONFIG.actions[aIdx]["cond"];
+    let cIdx = null;
+    let run = true;
+    for (cIdx in cond) {
+      if (typeof measurement[cIdx] === "undefined") run = false;
+      if (typeof cond[cIdx] === "object") {
+        if(cond[cIdx].cmp === ">" && measurement[cIdx] <= cond[cIdx].value) run = false; 
```

Comment:
Should we check if the `cpm` and `value` keys exist first before using them? Because these inputs we get from a client and they might be invalid